EXILED/Exiled.API/Features/Map.cs

```diff
@@ -74,6 +74,50 @@ public static class Map
             DecontaminationController.Singleton.NetworkDecontaminationOverride is DecontaminationController.DecontaminationStatus.Disabled ?
             DecontaminationState.Disabled : (DecontaminationState)DecontaminationController.Singleton._nextPhase;
 
+        /// <summary>
+        /// Gets the remaining time for the decontamination process.
+        /// </summary>
+        /// <returns>
+        /// The remaining time in seconds for the decontamination process.
+        /// Returns -1 if the controller is null or if decontamination is overridden.
+        /// Returns 0 if decontamination has not started yet.
+        /// </returns>
+        public static float RemainingDecontaminationTime
+        {
+            get
+            {
+                DecontaminationController controller = DecontaminationController.Singleton;
+                if (controller == null)
+                    return -1f;
+
+                if (controller.DecontaminationOverride != DecontaminationController.DecontaminationStatus.None)
+                    return -1f;
+
+                double currentTime = DecontaminationController.GetServerTime;
+                float totalTime = 0f;
+                bool decontaminationStarted = false;
+
+                for (int i = 0; i < controller.DecontaminationPhases.Length; i++)
+                {
+                    if (currentTime < controller.DecontaminationPhases[i].TimeTrigger)
+                    {
+                        if (!decontaminationStarted)
+                        {
+                            totalTime += (float)(controller.DecontaminationPhases[i].TimeTrigger - currentTime);
+                            decontaminationStarted = true;
+                        }
+                        else
+                        {
+                            totalTime += (float)(controller.DecontaminationPhases[i].TimeTrigger -
+                                                 controller.DecontaminationPhases[i - 1].TimeTrigger);
+                        }
+                    }
+                }
+
+                return decontaminationStarted ? totalTime : 0f;
+            }
+        }
```

Comment:
```suggestion
        public static float RemainingDecontaminationTime => DecontaminationController.Singleton.DecontaminationPhases[DecontaminationController.Singleton.DecontaminationPhases.Length - 1].TimeTrigger - DecontaminationController.GetServerTime;
```

Reply:
It's genius code!
But I'm not sure if this code actually works as expected.
The decontamination controller can be null or overridden.
When decontamination ends, this code could return negative values.
Still, this code is very clever.
We need to test this to verify it works as expected.

Reply:
you can still add the check you need